support/metrics/metricsTracker_test.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMergeEventProps(t *testing.T) {
+	commonProps := map[string]interface{}{
+		"1": 1,
+		"2": "two",
+	}
+	eventProps := map[string]interface{}{
+		"3": 3,
+		"4": "four",
+	}
+	wantMap := map[string]interface{}{
+		"1": 1,
+		"2": "two",
+		"3": 3,
+		"4": "four",
+	}
+
+	gotMap, e := mergeEventProps(commonProps, eventProps)
+	assert.Equal(t, wantMap, gotMap)
+	assert.NoError(t, e)
+}
+
+func TestToMapStringInterface(t *testing.T) {
+	success := map[string]interface{}{
```

Comment:
this input is not the most helpful since we convert from a map to a map. while this is a useful case to have, it's more interesting to test whether a struct can be converted to a map, especially since that's our use case. we can add that as a third case.

these three cases are fairly distinct, and we don't anticipate having many tests for which table based tests would make sense. It's probably better to separate out these cases into three different tests or as three subtests in this one test by using t.Run("success_map", ...), "success_struct" and then "failure_float"

Reply:
can you add the following types as values in the struct?:
bool, string, int, float, struct, []int